proposals/lip-0050.md

```diff
@@ -138,47 +223,126 @@ reclaimParamsSchema = {
 
 ##### Verification
 
-For a reclaim transaction `trs` to be valid, the legacy accounts substore must contain an entry with store key equal to `getLegacyAddress(trs.senderPublicKey)` and store value `{"balance": trs.params.amount}`, serialized using `legacyAccountsSchema`.
+```python
+def verify(trs: Transaction) -> None:
+    legacyAddress = getLegacyAddress(trs.senderPublicKey)
+    if legacyAccounts(legacyAddress) does not exists :
+        raise Exception('Public key does not correspond to a reclaimable account.')
 
-##### Execution
+    if legacyAccounts(legacyAddress).balance != trs.params.amount:
+        raise Exception('Input amount does not equal the balance of the legacy account.')
+```
 
-When a reclaim transaction `trs` is executed, the following is done:
+##### Execution
 
-* Delete the entry from the legacy accounts substore with store key `getLegacyAddress(trs.senderPublicKey)`.
-* Call the function `mint(newAddress, 0, trs.params.amount)`, where `newAddress` is the [20-byte address][lip-0018#addressComputation] derived from `trs.senderPublicKey`.
-  The function `mint` is defined in the [Token module][lip-0051#mint].
+```python
+def execute(trs: Transaction) -> None:
+    legacyAddress = getLegacyAddress(trs.senderPublicKey)
+    delete legacyAccounts(legacyAddress) from the legacy accounts substore
+
+    newAddress = 20-byte address derived from trs.senderPublicKey
+    # Unlock the tokens in the legacy reserve account and transfer them to the 
+    # account reclaiming the tokens.
+    token.unlock(ADDRESS_LEGACY_RESERVE, MODULE_ID_LEGACY, TOKEN_ID_LSK_MAINCHAIN, trs.params.amount)
+
+    token.transfer(ADDRESS_LEGACY_RESERVE, newAddress, TOKEN_ID_LSK_MAINCHAIN, trs.params.amount)
+
+    emitEvent(
```

Comment:
I think here we should drop the named fields and just write:

emitEvent(
        MODULE_ID_LEGACY,
        TYPE_ID_ACCOUNT_RECLAIM,
        serialize(
            {
                "legacyAddress": legacyAddress,
                "address":newAddress,
                "amount":trs.params.amount
            }, 
            accountReclaimedEventDataSchema
        ),
        [
            legacyAddress, 
            newAddress
        ]
    )

Reply:
Maybe you could point me to a reference that says that using named fields should be avoided?